### PR TITLE
CBR module implementation

### PR DESCRIPTION
**Pull request by whoffler**
### Description

CBR module implemententation

#### No release required

- [ ] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other changes that don't affect Terraform code

#### Release required

- [ ] Bug fix (patch release (`x.x.X`): Change that fixes an issue and is compatible with earlier versions)
- [x] New feature (minor release (`x.X.x`): Change that adds functionality and is compatible with earlier versions)
- [ ] Breaking change (major release (`X.x.x`): Change that is likely incompatible with previous versions)

##### Release notes content

Replace this text with information that users need to know about the bug fixes, features, and breaking changes. This information helps the merger write the commit message that is published in the release notes for the module.

---

### Checklist for reviewers

- [ ] The PR references a GitHub issue.
- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Merge by using "Squash and merge".
- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author.

    The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -94,67 +27,92 @@ unless real values don't help users know what to change.
 -->
 
 ```hcl
-
+module "ibm_cbr" "zone" {
+  # replace master with version
+  source = "git::https://github.com/terraform-ibm-modules/terraform-ibm-cbr//cbr-zone-module?ref=master"
```
Comment (ocofaigh) [inline README.md:32]: in public primary branch is `main`, so replace `master` with `main` here and in the comment above it

File: README.md [hunk 2]
```diff
@@ -94,67 +27,92 @@ unless real values don't help users know what to change.
...
+  name             = "zone_for_pg_access"
+  account_id       = data.ibm_iam_account_settings.iam_account_settings.account_id
+  zone_description = "Zone created from terraform"
+  addresses        = [{type  = "vpc",value = "vpc_crn"}]
+}
+
+module "ibm_cbr" "rule" {
+  # replace master with version
+  source = "git::https://github.com/terraform-ibm-modules/terraform-ibm-cbr//cbr-rule-module?ref=master"
```
Comment (ocofaigh) [inline README.md:41]: in public primary branch is `main`, so replace `master` with `main` here and in the comment above it

File: cbr-zone-module/README.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Default example
+
+Example to create a zone for Context Based Restrictions
```
Comment (ocofaigh) [inline cbr-zone-module/README.md:3]: this is not an example. This is an actual (sub)module (called by the root level module), so this readme should be titled `CBR Zone Module` and should describe what the module does

File: examples/multizone-rule/variables.tf [hunk 4]
```diff
@@ -27,3 +27,19 @@ variable "resource_tags" {
...
+variable "rule_description" {
+  type        = string
+  description = "(Optional, String) The description of the rule"
+  default     = null
+}
+
+variable "enforcement_mode" {
+  type        = string
+  description = "(String) The rule enforcement mode"
+  default     = "enabled"
```
Comment (ocofaigh) [inline examples/multizone-rule/variables.tf:44]: this value changed to `report` in a recent PR internally - please check that Pr to ensure all of those changes made it into this PR

File: variables.tf [hunk 5]
```diff
@@ -2,10 +2,99 @@
...
+      name  = "va"
+      value = "va"
+    }]
+  }]
+}
+
+variable "enforcement_mode" {
+  type        = string
+  description = "(String) The rule enforcement mode"
+  default     = "enabled"
```
Comment (ocofaigh) [inline variables.tf:71]: this value changed to `report` in a recent PR internally - please check that Pr to ensure all of those changes made it into this PR

File: main.tf [hunk 6]
```diff
@@ -1,3 +1,29 @@
-/********************************************************************
-This file is used to implement the ROOT module.
-*********************************************************************/
+##############################################################################
+# Get Cloud Account ID
+##############################################################################
+
+data "ibm_iam_account_settings" "iam_account_settings" {
```
Comment (ocofaigh) [inline main.tf:5]: we have documented that you need:
```
Account Management
   Editor role access
```
However I thought this data lookup required admin access (I could be wrong). Can we confirm?

File: tests/README.md [hunk 7]
```diff
@@ -2,7 +2,7 @@
 
 # Tests
 
-For information about how to create and run tests, see [Validation tests](https://terraform-ibm-modules.github.io/documentation/#/tests) in the project documentation.
+For information about how to create and run tests, see [Validation tests](https://github.ibm.com/GoldenEye/documentation/blob/master/tests.md) in the project documentation.
```
Comment (ocofaigh) [inline tests/README.md:5]: change this back to what it was in the template (its different from internal)

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see comments

File: README.md [hunk 8]
```diff
@@ -1,88 +1,21 @@
...
-# Terraform Modules Template Project
-<!-- UPDATE BADGE: Update the link for the following badge-->
-[![Incubating (Not yet consumable)](https://img.shields.io/badge/status-Incubating%20(Not%20yet%20consumable)-red)](https://terraform-ibm-modules.github.io/documentation/#/badge-status)
-[![Build status](https://github.com/terraform-ibm-modules/terraform-ibm-module-template/actions/workflows/ci.yml/badge.svg)](https://github.com/terraform-ibm-modules/terraform-ibm-module-template/actions/workflows/ci.yml)
+# Context Based Restrications Module
+<!-- UPDATE BADGES:
+1. Make sure that the badge link for the current status of the module is correct. For the status options, see https://github.ibm.com/GoldenEye/documentation/blob/master/status.md.
+2. Update the "Build Status" badge to point to the travis pipeline for the module. Replace "module-template" in two places.
+3. Update the "latest release" badge to point to the new module. Replace "module-template" in two places.
+-->
```
Comment (ocofaigh) [inline README.md:9]: Remove this ^^ It has links to internal

File: README.md [hunk 9]
```diff
@@ -94,67 +27,92 @@ unless real values don't help users know what to change.
 -->
 
 ```hcl
-
+module "ibm_cbr" "zone" {
+  # replace master with version
```
Comment (ocofaigh) [inline README.md:31]: master -> main

File: README.md [hunk 10]
```diff
@@ -94,67 +27,92 @@ unless real values don't help users know what to change.
...
+  source = "git::https://github.com/terraform-ibm-modules/terraform-ibm-cbr//cbr-zone-module?ref=main"
+  name             = "zone_for_pg_access"
+  account_id       = data.ibm_iam_account_settings.iam_account_settings.account_id
+  zone_description = "Zone created from terraform"
+  addresses        = [{type  = "vpc",value = "vpc_crn"}]
+}
+
+module "ibm_cbr" "rule" {
+  # replace master with version
```
Comment (ocofaigh) [inline README.md:40]: master -> main

**Review by ocofaigh — CHANGES_REQUESTED** [review]

File: tests/pr_test.go [hunk 11]
```diff
@@ -8,17 +8,32 @@ import (
 	"github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper/testhelper"
 )
 
-// Use existing resource group
 const resourceGroup = "geretain-test-resources"
```
Comment (ocofaigh) [inline tests/pr_test.go:11]: `geretain-test-cbr` is now created for this module, please use it

File: main.tf (hunk 6 above)
Comment (whoffler) [reply main.tf:5]: Editor access is adequate

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**